code-studio/src/js/progress.js

```diff
@@ -160,9 +160,11 @@ progress.renderStageProgress = function (stageData, progressData, clientProgress
 };
 
 progress.renderCourseProgress = function (scriptData) {
+  var teacherCourse = $('#landingpage').hasClass('teacher-course');
   var mountPoint = document.createElement('div');
   $('.user-stats-block').prepend(mountPoint);
   ReactDOM.render(React.createElement(CourseProgress, {
+    display: teacherCourse ? 'list' : 'dots',
```

Comment:
Agree, but this code will be ripped out in the next iteration.  Each puzzle will control how it's rendered based on whether it's a `level "Sample Puzzle"` or `resource "Sample Puzzle"` in the `.script` file.